### PR TITLE
Support load nid mapping from mapping files saved by dgl distpartition tools

### PR DESCRIPTION
**Pull request by classicsong**
The DGL distpartition tool saves the Graph node ID to Partition node ID mappings in a dgl object (https://github.com/dmlc/dgl/blob/eb43489397daf5506494d2cc5eaf7d7ff9dbefff/tools/distpartitioning/utils.py#L578-L583), which is not compatible with how gconstruct save node ID mappings.

#636 

This PR is going to support both id mappings saved by DGL distpartition and GraphStorm gconstruct.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by thvasilo** [comment]
This has already been tested with Gconstruct+SageMaker execution?

**Comment by classicsong** [comment]
> This has already been tested with Gconstruct+SageMaker execution?

I just tested it locally with GConstruct graph + reformat mapping files (I manually save them as .dgl).
I can test it with SageMaker.

**Comment by thvasilo** [comment]
Let's test on SM to ensure the required files are downloaded locally as expected.

**Comment by classicsong** [comment]
> Let's test on SM to ensure the required files are downloaded locally as expected.

I checked manually, it works.

File: tests/unit-tests/test_config.py [hunk 1]
```diff
@@ -1536,10 +1539,64 @@ def must_fail(conf):
                       "node_types": []}]
         must_fail(lm_config)
 
+def test_id_mapping_file():
+    import tempfile
```
Comment (thvasilo) [inline tests/unit-tests/test_config.py:1543]: Let's always put imports at the top of the file

**Review by thvasilo — APPROVED** [review]
LGTM since tests pass + SageMaker has been manually tested. One small change requested before pushing.

**Pull request merged**